api/src/main/java/org/pmiops/workbench/workspaces/WorkspacesController.java

```diff
@@ -1010,6 +1044,76 @@ static String notebookLockingEmailHash(String bucket, String email) {
     return ResponseEntity.ok(recentWorkspaceResponse);
   }
 
+  private WorkspaceResource createWorkspaceResourceFromWorkspace(DbWorkspace workspace) {
+    return new WorkspaceResource()
+        .workspaceFirecloudName(workspace.getFirecloudName())
+        .workspaceId(workspace.getWorkspaceId())
+        .workspaceNamespace(workspace.getWorkspaceNamespace())
+        .workspaceBillingStatus(workspace.getBillingStatus());
+  }
+
+  @Override
+  public ResponseEntity<WorkspaceResourceResponse> getDataPageItems(
+      String workspaceNamespace, String workspaceId) {
+    // This also enforces registered auth domain.
+    workspaceService.enforceWorkspaceAccessLevel(
+        workspaceNamespace, workspaceId, WorkspaceAccessLevel.READER);
+
+    DbWorkspace workspace =
+        workspaceService.getRequiredWithCohorts(workspaceNamespace, workspaceId);
+    Set<DbCohort> cohorts = workspace.getCohorts();
+    WorkspaceResourceResponse workspaceResources = new WorkspaceResourceResponse();
+    workspaceResources.addAll(
+        cohorts.stream()
+            .map(cohortMapper::dbModelToClient)
+            .map(
+                cohort ->
+                    createWorkspaceResourceFromWorkspace(workspace)
+                        .cohort(cohort)
+                        .modifiedTime(cohort.getLastModifiedTime().toString()))
+            .collect(Collectors.toList()));
+    // TODO: Convert cohorts to resources
```

Comment:
Can we bite this off all at once? How much does that blow up the ticket?

I'll be happy to help. I haven't actually written any backend code in like six weeks.

Reply:
That's a stale todo, this is already done :D